packages/studio-ui/src/AppWithLazyLoading.tsx

```diff
@@ -0,0 +1,75 @@
+import LoadingOverlay from "./components/LoadingOverlay";
+import { Suspense, lazy, useEffect, useState } from "react";
+import useStudioStore from "./store/useStudioStore";
+import ProgressBar from "./components/ProgressBar";
+import loadComponents from "./utils/loadComponents";
+import classNames from "classnames";
+
+const AppPromise = import("./App");
+const App = lazy(() => AppPromise);
+
+export default function AppWithLazyLoading() {
+  const [loadedCount, totalCount] = useStudioStore((store) => [
+    Object.keys(store.fileMetadatas.UUIDToImportedComponent).length,
+    Object.keys(store.fileMetadatas.UUIDToFileMetadata).length,
+  ]);
+  const componentsLoaded = loadedCount === totalCount;
+  const [appLoaded, setAppLoaded] = useState(false);
+
+  useEffect(() => {
+    loadComponents();
+    void AppPromise.then(() => setAppLoaded(true));
+  }, []);
+
+  return (
+    <LoadingOverlay
+      loading={!componentsLoaded || !appLoaded}
+      overlay={
+        <>
+          {renderComponentLoadingProgress(
+            loadedCount,
+            totalCount,
+            componentsLoaded
+          )}
+          {renderBundleMessage(appLoaded)}
+        </>
+      }
+    >
+      <Suspense>
```

Comment:
if we have the loading overlay, do we still need the suspense?  Or does it do something else other than display a fallback UI?

Reply:
is it just so everything in the app renders at once?